tests/src/3d/testqgs3dcameracontroller.cpp

```diff
@@ -1173,5 +1172,20 @@ void TestQgs3DCameraController::testRotationCenterRotationCameraRotationCenter()
   mapSettings->setLayers( {} );
 }
 
+void TestQgs3DCameraController::waitForNearPlane( QgsOffscreen3DEngine &engine, Qgs3DMapScene *scene, float atLeast ) //#spellok
+{
+  // XXX: Sometimes the near/far planes aren't calculated correctly, so they're
+  // left at the too-deep default. This causes the rest of the test to fail in
+  // weird ways every once in a while, so loop until we get good values.
+  do
+  {
+    // Force recalcualtion of near/far planes.
+    scene->cameraController()->mCameraChanged = true;
+
+    // this call is not used but ensures to synchronize the scene
+    Qgs3DUtils::captureSceneImage( engine, scene );
+  } while ( scene->cameraController()->camera()->nearPlane() < atLeast ); //#spellok
```

Comment:
May be you could add a max iteration check too avoid infinite loop or too long wait.